### ISSUE TITLE
Making an Open edX Commons Learning Object Repository

### ISSUE THREAD

**Issue by ormsbee** [openedx/openedx-learning#23]
Assume the following:

* We want to start by focusing on the small-Item reusability use case.
* We've split the concepts of [Learning Package and Learning Context](https://github.com/openedx/openedx-learning/issues/21).
 
What would a centralized commons look like, and what's the easiest way it could work for end users?

# Executing in-place

A number of institutions like MIT operate multiple instances and have difficulty managing their content across those instances, leading to various solutions over time. Instead of trying to facilitate content replication across instances, what if we serve content centrally? Not in the sense that there's only ever one server, but in the sense that an organization could choose to manage all their content in one place.

So what could that look like? Some possibilities:

* An LTI Advantage interface would be a basic requirement.
* We could expose a separate iframe-based API for the anonymous-user use case. This would drastically lower the burden of using very small pieces of content in random places. We would want to have some API layer over this to do things like iframe-resize messages, so this could likely be encapsulated into a web component.
* The authoring interface is in this new web application.
* There would be a runtime dependency between LMS instances and one or more Commons instances.

**Comment by jmakowski1123** [comment]
Couple thoughts:

- I agree that we'd need to minimize having authors bounce between Studio and one or more Commons. I think we can design solutions and workflows to mitigate this. For example, similar to the way we're thinking about baking a "menu" of LTI apps into Studio, we could consider the same here - ie, baking a search/discovery layer into the Studio authoring flow that would enable authors an easy flow to access their content.
- Instead of thinking about one Commons per institution, what if there were one central Open edX Commons, with levels of access restrictions at the institution level? For example, MIT wants to make Batch A of sequences share-able only across MIT instances, but they are happy to make Batch B of sequences available for any Open edX author to use in any instance.
- Both bullet points above reinforce the need to continue fleshing stories around descriptive metadata, as that will be the key to ensuring content is searchable and discoverable in a meaningful way.

**Comment by ormsbee** [comment]
> Instead of thinking about one Commons per institution, what if there were one central Open edX Commons, with levels of access restrictions at the institution level? For example, MIT wants to make Batch A of sequences share-able only across MIT instances, but they are happy to make Batch B of sequences available for any Open edX author to use in any instance.

Yeah, I think we'd want to operate a big site that multiple institutions could use. But the software being open source, I imagine that the end state is going to have a few different instances in different places, and that it wouldn't work to assume a 1:1 mapping of Commons instances to Open edX Studio or LMS instances.

> I'm not sure we can (or want) to call this thing a Commons. For further thought -

Yeah, I'm definitely open to other names on this one. I've thought of a half dozen others and even the most ridiculous of those was taken. 😛

**Comment by ormsbee** [comment]
Just replaced Commons with LOR for now, even though this is a bit more expansive than what a LOR would traditionally be.
